examples/keras_recipes/reproducibility_recipes.py

```diff
@@ -0,0 +1,274 @@
+"""
+Title: Reproducibility in Keras Models
+Author: [Frightera](https://github.com/Frightera)
+Date created: 2023/05/05
+Last modified: 2023/05/05
+Description: Demonstration of random weight initialization and reproducibility in Keras models.
+Accelerator: GPU
+"""
+"""
+## Introduction
+This example demonstrates how to control randomness in Keras models. Sometimes
+you may want to reproduce the exact same results across runs, for experimentation
+purposes or to debug a problem.
+
+This tutorial applies to Keras 2.7 and higher.
+"""
+
+"""
+## Setup
+"""
+import inspect
+import json
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+# Set the seed using keras.utils.set_random_seed. This will set:
+# 1) `numpy` seed
+# 2) `tensorflow` random seed
+# 3) `python` random seed
+keras.utils.set_random_seed(812)
+
+# This will make TensorFlow ops as deterministic as possible.
+tf.config.experimental.enable_op_determinism()
+
+"""
+## Weight initialization in Keras
+
+Most of the layers in Keras have `kernel_initializer` and `bias_initializer`
+parameters. These parameters allow you to specify the strategy used for
+initializing the weights of layer variables. The following built-in initializers
+are available as part of `tf.keras.initializers`:
+"""
+to_be_removed = ["Initializer", "Ones", "Zeros", "Identity"]
+initializers_list = [
+    string
+    for string in dir(keras.initializers)
+    if string[0].isupper() and string not in to_be_removed
+]
+print(initializers_list)
+
+# Let's call each initializer two times and store the results in a dictionary.
+results = {}
+
+for initializer_name in initializers_list:
+    print(f"Running {initializer_name} initializer")
+    # Get the initializer object from the Keras initializers module
+    initializer = getattr(keras.initializers, initializer_name)
+    results[initializer_name] = []
+
+    # Get the signature of the initializer
+    initializer_signature = inspect.signature(initializer)
+
+    for _ in range(2):
+        # In order to get same results across multiple runs from an initializer,
+        # you need to specify a seed value. Note that this is not related to
+        # keras.utils.set_random_seed or tf.config.experimental.enable_op_determinism.
+        # If you comment those lines, you will still get the same results.
+        if "seed" in initializer_signature.parameters:
+            result = initializer(seed=42)(shape=(3, 3))
+        else:
+            result = initializer()(shape=(3, 3))
+        results[initializer_name].append(result)
+
+# Check if the results are equal.
+all_equal = True
+for initializer_name, initializer_results in results.items():
+    if not tf.experimental.numpy.allclose(
```

Comment:
so we're not quite removing modules, we're just checking to ensure that these output identical results?  I think that is reasonable!